From ce020579e066bb2283c140806102845aabf6b362 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:02 -0400
Subject: [scsi] qla2xxx: Clear RISC INT reg only for an event and not always while polling

Message-id: <1408391817-6389-15-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87913
O-Subject: [PATCH RHEL 7.1 e-stor 14/69] qla2xxx: Clear RISC INT reg only for an event and not always while polling.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Atul Deshmukh <atul.deshmukh@qlogic.com>

scsi commit id 02a9ae6e31975e97777a4e3db6e33e059f256c05

Signed-off-by: Atul Deshmukh <atul.deshmukh@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_nx.c b/drivers/scsi/qla2xxx/qla_nx.c
index 4822533..1656a21 100644
--- a/drivers/scsi/qla2xxx/qla_nx.c
+++ b/drivers/scsi/qla2xxx/qla_nx.c
@@ -2224,8 +2224,8 @@ qla82xx_poll(int irq, void *dev_id)
        stat * 0xff);
    break;
   }
+  WRT_REG_DWORD(&reg->host_int, 0);
  }
- WRT_REG_DWORD(&reg->host_int, 0);
  spin_unlock_irqrestore(&ha->hardware_lock, flags);
 }
 
-- 
1.7.1